File: VNCore/Configuration.cs
using System;
using System.IO;
using System.Xml;

namespace VNCore
{
    public static class Configuration
    {
        public static string UserName { get; set; }
        public static string Language { get; set; }
        public static bool Fullscreen { get; set; }
        public static bool AutoSave { get; set; }
        public static int SoundVolume { get; set; }
        public static void Init()
        {
            if (File.Exists(EnvironmentVariables.ConfigurationFile))
                Load();
            else
            {
                Fullscreen = true;
                Save();
            }
        }
        public static void Save()
        {
            using (var writer = XmlWriter.Create(EnvironmentVariables.ConfigurationFile))
            {
                writer.WriteStartElement("Config");
                writer.WriteAttributeString("UserName", UserName ?? (UserName = Environment.UserName));
                writer.WriteAttributeString("Language", Language ?? (Language = System.Globalization.CultureInfo.CurrentCulture.TwoLetterISOLanguageName));
                writer.WriteAttributeString("Fullscreen", Fullscreen.ToString());
                writer.WriteAttributeString("AutoSave", AutoSave.ToString());
                writer.WriteAttributeString("SoundVolume", SoundVolume.ToString());
                writer.WriteEndElement();
            }
        }
        public static void Load()
        {
            if (File.Exists(EnvironmentVariables.ConfigurationFile))
                using (var reader = XmlReader.Create(EnvironmentVariables.ConfigurationFile))
                    while (reader.Read())
                        if (reader.IsStartElement("Config"))
                        {
                            UserName = reader.GetAttribute("UserName");
                            Language = reader.GetAttribute("Language");
                            Fullscreen = bool.Parse(reader.GetAttribute("Fullscreen"));
                            AutoSave = bool.Parse(reader.GetAttribute("AutoSave"));
                            SoundVolume = int.Parse(reader.GetAttribute("SoundVolume"));
                        }
        }
    }
}

File: VNCore.Java/Configuration.java
import com.sun.org.apache.xerces.internal.impl.PropertyManager;
import com.sun.org.apache.xerces.internal.impl.XMLStreamReaderImpl;
import com.sun.xml.internal.stream.writers.XMLStreamWriterImpl;

import javax.xml.stream.XMLStreamException;
import javax.xml.stream.XMLStreamReader;
import javax.xml.stream.XMLStreamWriter;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class Configuration {
    public static String userName;
    public static String language;
    public static Boolean fullscreen;
    public static Boolean autoSave;
    public static Integer soundVolume;

    public static void init() throws IOException, XMLStreamException {
        if (Files.exists(Paths.get(EnvironmentVariables.configurationFile)))
            load();
        else {
            fullscreen = true;
            save();
        }
    }

    public static void save() throws XMLStreamException, IOException {
        XMLStreamWriter writer = new XMLStreamWriterImpl(new FileOutputStream(EnvironmentVariables.configurationFile), "UTF-8", new PropertyManager(PropertyManager.CONTEXT_WRITER));
        writer.writeStartElement("Config");
        writer.writeAttribute("UserName", userName);
        writer.writeAttribute("Language", language);
        writer.writeAttribute("Fullscreen", fullscreen.toString());
        writer.writeAttribute("AutoSave", autoSave.toString());
        writer.writeAttribute("SoundVolume", soundVolume.toString());
        writer.writeEndElement();
    }

    public static void load() throws XMLStreamException, IOException {
        if (Files.exists(Paths.get(EnvironmentVariables.configurationFile))) {
            XMLStreamReader reader = new XMLStreamReaderImpl(new FileInputStream(EnvironmentVariables.configurationFile), "UTF-8", new PropertyManager(PropertyManager.CONTEXT_WRITER));
            while (reader.hasNext())
                if (reader.isStartElement() && reader.getLocalName().equals("Config")) {
                    String namespaceUri = reader.getNamespaceURI();
                    userName = reader.getAttributeValue(namespaceUri, "UserName");
                    language = reader.getAttributeValue(namespaceUri, "Language");
                    fullscreen = Boolean.parseBoolean(reader.getAttributeValue(namespaceUri, "Fullscreen"));
                    autoSave = Boolean.parseBoolean(reader.getAttributeValue(namespaceUri, "AutoSave"));
                    soundVolume = Integer.parseInt(reader.getAttributeValue(namespaceUri, "SoundVolume"));
                    reader.next();
                }
        }
    }

}